harvesters/src/main/java/org/fao/geonet/kernel/harvest/harvester/webdav/Harvester.java

```diff
@@ -292,6 +288,11 @@ private void addMetadata(RemoteFile rf) throws Exception {
             md = translateMetadataContent(context, md, schema);
         }
 
+        if (StringUtils.isNotEmpty(params.xslfilter)) {
+            md = HarvesterUtil.processMetadata(dataMan.getSchema(schema),
+                md, processName, processParams);
```

Comment:
```suggestion
                md, processName, processParams);
                
            schema = dataMan.autodetectSchema(md);
```

It is always good to update schema if XSLT process change it. 
eg. https://github.com/geonetwork/core-geonetwork/blob/main/harvesters/src/main/java/org/fao/geonet/kernel/harvest/harvester/csw/Aligner.java#L312-L314